cegui/src/falagard/WidgetComponent.cpp

```diff
@@ -188,10 +188,11 @@ namespace CEGUI
     void WidgetComponent::removePropertyInitialiser(const String& name)
     {
         for(PropertiesList::iterator i = d_properties.begin();
-                i < d_properties.end();
-                ++i)
+                i != d_properties.end();)
```

Comment:
Valid fix to me. The typical erase-while-iterate in C++.

Reply:
Would have liked to apply the Erase–remove idiom, but if it's usually only going to find once instance there's no point in being more efficent.